torch_xla/csrc/runtime/cache.h

```diff
@@ -106,6 +122,107 @@ class Cache {
   ElementMap element_map_;
 };
 
+// A persistent cache which serializes values to disk. This wraps a Cache
+// instance, so values will only be read from disk once and subsequent reads
+// will go through the wrapped Cache.
+template <typename K, typename T, typename H = std::hash<K>,
+          typename E = std::equal_to<K>>
+class PersistentCache : public AbstractCache<K, T, H, E> {
```

Comment:
Can we specialize the PersistentCache a bit more? Do we really need to carry that many template parameters?

Reply:
I don't see a use case for the H and E templates, so those can be dropped and the default value directly applied the parent class. Keeping the same interface as Cache lets this be a drop-in replacement though, maybe we can follow up to remove the unnecessary templates from both?

Reply:
Let me take a closer look (probably after my break...). I need to play around with the patch to awake my C++ muscles...